kaiax/staking/impl/getter.go

```diff
@@ -100,7 +100,7 @@ func (s *StakingModule) getFromStateByNumber(num uint64) (*staking.StakingInfo,
 // Efficiently read addresses and balances from the AddressBook in one EVM call.
 // Works by temporarily injecting the MultiCallContract to a copied state.
 func (s *StakingModule) getFromState(header *types.Header, statedb *state.StateDB) (*staking.StakingInfo, error) {
-	isPrague := s.ChainConfig.IsPragueForkEnabled(header.Number)
+	isForPrague := s.ChainConfig.IsPragueForkEnabled(new(big.Int).Add(header.Number, common.Big1))
```

Comment:
I think adding one is to "reverse" `sourceNum` to `num` from `GetStakingInfo(num)`, given that `sourceNum = num - 1`.
Wouldn't it be safer to pass rules so that `GetStakingInfo(num) -> getFromStateByNumber(sourceNum, rules) -> getFromState(sourceNum, rules)`?

Reply:
That would be better I think. But we still need to add one in `PreloadFromState` anyway. How do you think about it?

Reply:
Maybe introducing a reversal method like `requestBlockNum` would be more readable?

Reply:
Oh, in that case, let's leave it as-is. I was worried that in the far future `sourceNum` calculation can change. Reversing may not always be deterministic, such as pre-Kaia.